### PR TITLE
Fixes #1022 by making starlette an optional dependency.

### LINKED ISSUE #1022: Make `starlette` an optional dependency
Since `starlette` is only used for the debug server it should be an optional dependency in `pyproject.toml` like e.g. `uvicorn`.

### PR DESCRIPTION
**Pull request by mawiegand**
## Description

Fixes #1022 by making `starlette` an optional dependency in the `pyproject.toml` file.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* #1022

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Release type: patch
+
+Fixes #1022 by making starlette an optional dependency.
```
Comment (patrick91) [inline RELEASE.md:3]:
```suggestion
Fixes [#1022](https://github.com/strawberry-graphql/strawberry/issues/1022) by making starlette an optional dependency.
```

**Review by patrick91 — APPROVED** [review]
Looks good to me! Maybe in future we can also ask to install starlette when trying to start the debug server 😊

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

Fixes [#1022](https://github.com/strawberry-graphql/strawberry/issues/1022) by making starlette an optional dependency.

**Comment by patrick91** [comment]
@mawiegand thanks for your first contribution :) I've invited you to our organisation!

Feel free to request a sticker here -> https://forms.gle/dmnfQUPoY5gZbVT67

**Pull request merged**